proxy/handler_test.go

```diff
@@ -0,0 +1,60 @@
+package proxy_test
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/ONSdigital/dp-legacy-cache-proxy/config"
+	"github.com/ONSdigital/dp-legacy-cache-proxy/proxy"
+	"github.com/gorilla/mux"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+var babbageEndpoint = "/economy/environmentalaccounts/bulletins/ukenvironmentalaccounts/2013-06-26"
+
+func TestProxyHandler(t *testing.T) {
+	Convey("Given a Proxy Handler ", t, func() {
```

Comment:
This is called "Given a Proxy Handler", but that's not really what's going on here, you've only defined a router and a context. In any case, when writing tests in the Given/When/Then format, you should try to organise them so that the setup happens in the Given block (for instance, you could move the creation of the `legacyProxy` variable here), the execution happens in the When block, and the assertions happens in the Then block. This isn't a hard rule, sometimes it may be necessary to mix things a bit, or it may be unpractical to be that rigid, but in general you should adhere to this, as it's a well known convention.

Reply:
Yes, this makes sense and that's how I had it originally, but due to needing a different configuration in each proxy setup for mocking the babbage server I moved this bit of the code within the Given blocks